R/chromote_session.R

```diff
@@ -39,24 +39,26 @@ ChromoteSession <- R6Class(
       # Create a session from the Chromote. Basically the same code as
       # new_session(), but this is synchronous.
       if (is.null(targetId)) {
-        target <- parent$Target$createTarget(
+        p <- parent$Target$createTarget(
           "about:blank",
           width = width,
           height = height,
           wait_ = FALSE
-         )
+         )$
+          then(function(value) {
+            parent$Target$attachToTarget(value$targetId, flatten = TRUE, wait_ = FALSE)
+          })$
+          then(function(value) {
+            private$session_id <- value$sessionId
+            self$parent$register_session(self)
+          })
       } else {
-        target <- promise_resolve(list(targetId = targetId))
+        p <- parent$Target$attachToTarget(targetId, flatten = TRUE, wait_ = FALSE)$
```

Comment:
Can you add a comment about this explaining why it's like this, and what it should be changed to once we get the bug fixed?

Reply:
Although, I guess we could leave it like this, so maybe the comment isn't necessary. If so, I'd prefer that the `then` with `value$session_id` is moved out of the `if-else` so that it's shared between the two code paths.